### DISCUSSION TITLE
Errors in cmake configuration: could not find git for clone of cxxopts-populate

### DISCUSSION THREAD

**Discussion by zbounik** [sofa-framework/sofa#2716] [Build / Config / Environment] [answered]
Hi, 

Trying to build SOFA 21.12 stable on Ubuntu 20.04, I get these persisting errors:

CMake Error at /usr/local/share/cmake-3.18/Modules/ExternalProject.cmake:2350 (message):
  error: could not find git for clone of cxxopts-populate
Call Stack (most recent call first):
  /usr/local/share/cmake-3.18/Modules/ExternalProject.cmake:3206 (_ep_add_download_command)
  CMakeLists.txt:13 (ExternalProject_Add)

-- Configuring incomplete, errors occurred!
See also "/home/za/SOFA/build/_deps/cxxopts-subbuild/CMakeFiles/CMakeOutput.log".

CMake Error at /usr/local/share/cmake-3.18/Modules/FetchContent.cmake:975 (message):
  CMake step for cxxopts failed: 1
Call Stack (most recent call first):
  /usr/local/share/cmake-3.18/Modules/FetchContent.cmake:1082:EVAL:2 (__FetchContent_directPopulate)
  /usr/local/share/cmake-3.18/Modules/FetchContent.cmake:1082 (cmake_language)
  modules/SofaGuiCommon/CMakeLists.txt:18 (FetchContent_Populate)

There is no cmake-3.18 in "/usr/local/share/". I reinstalled it and then upgraded it to cmake 3.22, but when I the problem persists. I appreciate any clue on this.

Best,
Zahra

**Comment by fredroy** [comment, marked as answer]
Hello @zbounik 
`error: could not find git for clone of cxxopts-populate`

It seems cmake did not find `git` on your system... Maybe a configuration problem ? (not well installed, not in `PATH`)

**Reply by zbounik** [reply]
Hi @fredroy 

Thanks yes it was both because of not installing the git. Installing and defining paths all from the terminal helps to avoid many errors.
